source/common/filesystem/posix/filesystem_impl.h

```diff
@@ -16,8 +16,14 @@ class FileImplPosix : public FileSharedImpl {
   ~FileImplPosix() override;
 
 protected:
+  struct FlagsAndPerm {
```

Comment:
Does it make sense to put FlagsAndPerm in a shared interface, rather than repeating it from win32 and posix?

Reply:
Because the translateFlags method is platform specified function ( the `translateFlags` is in the platform class not in shared interface). So the implementation should have their own flags and perm or other something more than this two flags.

But currently we only care about the perm and flags, so I'm ok with that . And will change.

Reply:
s/perm/mode/ and s/int/mode_t/ for the mode arg?